ui/src/lib/api-client/fms-client.ts

```diff
@@ -15,3 +19,66 @@ const authMiddleware: Middleware = {
 	}
 };
 fmsClient.use(authMiddleware);
+
+// Our swagger specs have some parameters defined as query parameters, but also include those parameters as
+// part of the path. Those path parameters cannot be replaced correctly, so we will need to strip them out.
+const fixPathsMiddleware: Middleware = {
+	async onRequest({ request }) {
+		// Strip any path parameters that haven't been provided
+		// E.g. /teamIssues/{noteId} becomes /teamIssues
+		let fixedUrl = request.url.replace(/\/%7B\w*%7D/g, '');
+		return new Request(fixedUrl, request);
+	}
+};
+fmsClient.use(fixPathsMiddleware);
+
+export type TeamIssue = components['schemas']['TeamIssueModel'];
+export interface FetchTeamIssueOptions {
+	noteId?: string;
+	teamNumber?: number;
+	issueType?: string;
+	resolutionStatus?: string;
+}
+export async function getTeamNotes(
+	fetch: any,
```

Comment:
What is fetch?

Reply:
Still getting used to Typescript and figuring the patterns out - I ended up not using the interface

Reply:
So that's a great question and I'm really not familiar enough with Svelte and JS to talk particularly eloquently about it, but at a high level Svelte [defines its own `fetch`](https://kit.svelte.dev/docs/load#making-fetch-requests) that provides some additional functionality on top of the native `fetch` API. To use that with `openapi-fetch`, we need to pass it along from the Svelte `load()` function.

That pattern is the recommendation from `openapi-fetch`:
https://openapi-ts.dev/openapi-fetch/examples#svelte-sveltekit

I am _not_ using it in some places where we aren't using a Svelte `load()` function, mostly because I also just don't know how to get access to Svelte's `fetch` in those cases.

It is something I want to dig into a bit more too, if only for my own learning